src/core/context/AppContextProvider.tsx

```diff
@@ -543,12 +546,39 @@ async function syncWithLocalStorage(
   );
 
   const existingConversation = allConversations[conversationID];
+
+  let tokensRemaining = contextLength;
+
+  const isDeepseekModel = id.includes('deepseek');
+  const isGPTModel = id.includes('gpt');
+
+  //  todo - refactor to use modelConfig for both and simplify conditionals
+  if (isDeepseekModel) {
+    //  If we get a response with usage, use that to calculate
+  } else {
+    //  todo - plug in deepseek approximation function
+  }
+
+  //  type guard won't be necessary when all model configs have a contextLimitMonitor
+  if (isGPTModel && modelConfig.contextLimitMonitor) {
+    try {
+      const metrics = await modelConfig.contextLimitMonitor(
+        messages,
+        contextLength,
+      );
+      tokensRemaining = metrics.tokensRemaining;
+    } catch {
+      //  estimate the amount
+    }
+  }
+
```

Comment:
This will get simplified after this when I rebase the deepseek branch and both models have a `contextLimitMonitor` function. 

We will call the model specific function and if it fails, we'll do a rough estimate based on character length